### ISSUE TITLE
Error building spark kernel client 

### ISSUE THREAD

**Issue by dimple2015** [ibm-et/spark-kernel#148]
Hi,
From the Top level project folder i.e spark-kernel, I was able to build with Spark 1.4.1 by running APACHE_SPARK_VERSION="1.4.0" sbt compile
APACHE_SPARK_VERSION="1.4.0" sbt pack
and I'm able to build start spark kernel successfully.
However when I go to spark-kernel/client dir and do sbt compile, I see 252 errors some which are below:

error] /root/spark-kernel/client/src/main/scala/com/ibm/spark/comm/ClientCommManager.scala:19: object annotations is not a member of package com.ibm.spark
[error] import com.ibm.spark.annotations.Experimental
[error]                      ^
[error] /root/spark-kernel/client/src/main/scala/com/ibm/spark/comm/ClientCommManager.scala:21: object KMBuilder is not a member of package com.ibm.spark.kernel.protocol.v5
[error] import com.ibm.spark.kernel.protocol.v5.{KMBuilder, UUID}
[error]        ^
[error] /root/spark-kernel/client/src/main/scala/com/ibm/spark/comm/ClientCommManager.scala:36: not found: type CommManager
[error] ) extends CommManager(commRegistrar)
[error]           ^
[error] /root/spark-kernel/client/src/main/scala/com/ibm/spark/comm/ClientCommManager.scala:31: not found: type Experimental
[error] @Experimental
[error]  ^
[error] /root/spark-kernel/client/src/main/scala/com/ibm/spark/comm/ClientCommManager.scala:34: not found: type KMBuilder
[error]   private val kmBuilder: KMBuilder,
[error]                          ^
[error] /root/spark-kernel/client/src/main/scala/com/ibm/spark/comm/ClientCommManager.scala:35: not found: type CommRegistrar
[error]   private val commRegistrar: CommRegistrar
[error]                              ^
[error] /root/spark-kernel/client/src/main/scala/com/ibm/spark/comm/ClientCommManager.scala:32: too many arguments for constructor Object: ()Object
[error] class ClientCommManager(
[error]                        ^
[error] /root/spark-kernel/client/src/main/scala/com/ibm/spark/comm/ClientCommManager.scala:45: not found: type CommWriter
[error]   override protected def newCommWriter(commId: UUID): CommWriter =

Am I missing something here? Pls. help

Thanks
Dimple

**Comment by chipsenkbeil** [comment]
Did you try building the client from the root of the Spark Kernel project? I think it needs to be done that way since the client project brings in other project code and wouldn't know about it from the root of the client module.

```bash
APACHE_SPARK_VERSION="1.4.0" sbt client/compile
```

**Comment by dimple2015** [comment]
Thanks Chip. That worked. I tried the https://github.com/ibm-et/spark-kernel/blob/master/client/src/test/scala/examples/ScalaSparkClientUsage.scala and modified the path to the json file according to my path(gave full path) as:
val profile: File = new File(getClass.getResource("/root/spark-kernel/client/src/test/resources/kernel-profiles/IOPubIntegrationProfile.json").toURI)
but got a null pointer exception even though I can view the file in that path. However when I created a Json string instead and used the ConfigFactory.parseString that worked. 
Not sure what's wrong in the getClass.getResource.?

Another question is how can I access the spark context or spark sql context in order to use spark , spark sql: methods such as sc.textfile()? If you would like me to open a separate issue for this , pls. let me know. Thanks

Dimple
